service/src/client.rs

```diff
@@ -118,6 +118,12 @@ pub trait ExecuteWithClient {
 			Client: AbstractClient<Block, Backend, Api = Api> + 'static;
 }
 
+/// Yet Another ExecuteWithClient
+pub trait YaExecuteWithClient {
```

Comment:
Maybe `ClientHandle`?

With a much better documentation on what it does and why we need it? Something like:
```
/// A handle to a Polkadot client instance
///
/// The Polkadot service supports multiple different runtimes, like
/// Westend, Polkadot itself, etc. As each runtime has a specialized
/// client, we need to hide them behind a trait. This is this trait. 
///
/// When wanting to work with the inner client, you need to use [`execute_with`](ClientHandle::execute_with).
/// See [`ExecuteWithClient`] for more information.
```

Reply:
omg I was not expecting that much 😆 :pray: